source/baseObject.py

```diff
@@ -72,20 +88,36 @@ def __init__(self,name,bases,dict):
 				else:
 					cache=cacheByDefault
 
+			abstract=dict.get('_abstract_%s'%x,False)
 			if g and not s and not d:
-				setattr(self,x,(CachingGetter if cache else Getter)(g))
+				attr = (CachingGetter if cache else Getter)(g,abstract)
 			else:
-				setattr(self,x,property(fget=g,fset=s,fdel=d))
+				attr = (abstractproperty if abstract else property)(fget=g,fset=s,fdel=d)
+			if abstract:
+				newAbstractProps.add(x)
+			elif x in self.__abstractmethods__:
+				oldAbstractProps.add(x)
+			setattr(self,x, attr)
+
+		if newAbstractProps or oldAbstractProps:
+			# The __abstractmethods__ set is frozen, therefore we ought to override it.
+			self.__abstractmethods__=(self.__abstractmethods__|newAbstractProps)-oldAbstractProps

```

Comment:
Would it be equivelent to instead do a set intersection with newAbstratProps & __abstract__ eleminating oldAbstractprops?

Reply:
You mean, just self.__abstractmethods__ = (self.__abstractmethods__&newAbstractProps) . That would also eliminate abstract methods that aren't abstract properties at all.